tests/test_grading.py

```diff
@@ -92,9 +92,9 @@ def test_course_submissions_optimized(self):
             print("Running with {}".format(db.engine.name))
             query = self.assignment.mysql_course_submissions_query()
             data = [d for d in query]
-            backups_from_query = [d[6] for d in data if d[6]]
+            backups_from_query = [d[7] for d in data if d[7]]
```

Comment:
`course_submissions_query` does return a sequence of dict - but the mysql_query fn doesn't. I'd be okay with moving the creation of the sequence of dict into the `mysql_course_submissions_query` fn. 

Ideally we'd be using namedtuple objects and then we wouldn't have to worry about this.